sonar-db/src/main/java/org/sonar/db/version/v54/InsertGateAdminPermissionForEachProfileAdmin.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * SonarQube :: Database
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.db.version.v54;
+
+import java.sql.SQLException;
+import org.sonar.db.Database;
+import org.sonar.db.version.BaseDataChange;
+import org.sonar.db.version.MassUpdate;
+import org.sonar.db.version.Select;
+import org.sonar.db.version.SqlStatement;
+
+/**
+ * Global permission 'profileadmin' is split into 'profileadmin' and 'gateadmin' so for each permission in DB
+ * 'profileadmin', a permission 'gateadmin' must be inserted in DB to keep the same level as before this split is
+ * introduced in SQ.
+ */
+public class InsertGateAdminPermissionForEachProfileAdmin extends BaseDataChange {
+
+  public InsertGateAdminPermissionForEachProfileAdmin(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    updateGroupAnyOne(context);
+    updateOtherGroups(context);
+    updateUsers(context);
+  }
+
+  private void updateGroupAnyOne(Context context) throws SQLException {
+    MassUpdate update = context.prepareMassUpdate().rowPluralName("Group AnyOne");
+    update.select("select " +
+      "gr1.id " +
+      "from group_roles gr1 " +
+      "where " +
+      "gr1.role = 'profileadmin' " +
+      "and gr1.resource_id is null " +
+      "and gr1.group_id is null " +
+      "and not exists (" +
+      " select gr2.id " +
+      " from group_roles gr2 " +
+      " where " +
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal " where " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1192)